library/systemd/src/lib/yast2/service_configuration.rb

```diff
@@ -0,0 +1,194 @@
+    #   service does not support reload or does not run, then restart is used.
+    #
+    # @example three services
+    #   config = Yast2::ServiceConfiguration.new(s1, s2, s3)
+    def initialize(*services)
+      if services.any? { |s| !s.is_a?(Yast::SystemdServiceClass::Service) }
+        raise ArgumentError, "Services can be only Systemd Service - #{services.inspect}"
+      end
+
+      @services = services
+    end
+
+    # reads system status of services. Can be also used to reread current status.
+    # @raise <Yast::SystemctlError> when read services state failed
+    def read
+      # safe initial state when exception happen
+      @status = :unknown
+      @autostart = :unknown
+
+      read_status
+      read_autostart
+    end
+
+    # writes services new status
+    # @raise <Yast::SystemctlError> when set service to target state failed
+    def write
+      write_action
+      write_autostart
+    end
+
+    # returns current running status, but when read failed return `:unknown`.
+    # @return [:active, :inactive, :inconsistent, :unknown] current status of
+    #   services:
+    #
+    #   - `:active` when all services are active
+    #   - `:inactive` when all services are inactive
+    #   - `:inconsistent` when part of services is active and part not.
+    #       Can happen only when configuration handle multiple services
+    #   - `:unknown` when read of current status failed.
+    def status
+      read unless @status
+
+      @status
+    end
+
+    # returns currently set target action. If it is not yet set it returns
+    # `:nothing`.
+    # @return [:start, :stop, :restart, :nothing] action for all services:
+    #
+    #   - `:start` to start all services. If service is already active, do nothing.
+    #   - `:stop`  to stop all services. If service already is inactive, do nothing.
+    #   - `:restart` restart all services. If service is inactive, it is started.
+    #   - `:reload` reload all services that support it and restart that does not
+    #       support it. If service is inactive, it is started.
+    #   - `:nothing` do not touch anything.
+    def action
+      return :nothing unless @action
+
+      @action
+    end
+
+    ACTIONS = [ :start, :stop, :reload, :restart, :nothing].freeze
+    # sets target action for services
+    # @param action[:start, :stop, :restart, :reload, :nothing] for possible values and
+    #  its explanation please see return value of {target_action}
+    def action= (action)
+      if !ACTIONS.include?(action)
+        raise ArgumentError, "Invalid parameter #{action.inspect}"
+      end
+
+      @action = action
+    end
+
+    # returns currently set autostart configuration. If it is not yet set it
+    # calls read and return current system state.
+    # @return [:on_boot, :on_demand, :manual, :inconsistent, :unknown] autostart
+    #   configuration:
+    #
+    #   - `:on_boot` all services start during boot.
```

Comment:
I know it is a different thing, but what about using `:boot` instead of `:on_boot` and `:demand` instead of `:on_demand` to keep consistency? Or just the other way around: I could adjust the `SystemdService` API.